### PR TITLE
Fix Coverity Issues for battery Module.

### PR DESCRIPTION
**Pull request by vilasrk**
Tracked-On: OAM-105560

### CODE REVIEW

File: vm_battery_utility/battery_sysfsread.c [hunk 1]
```diff
@@ -87,10 +87,20 @@ int read_sysfs_values(char *filename, void *buf, int len, int flag)
 		return -ENODEV;
 	}
 
-	if (flag==0)
-		fread(buf, len, 1, fp);
-	else
-    		fscanf (fp, "%d", (int*)buf);  /* read/validate value */
+	if (flag==0) {
+	    if(fread(buf, len, 1, fp) == EOF)
+		{
```
Comment (JaikrishnaNemallapudi) [inline vm_battery_utility/battery_sysfsread.c]: check space indentation

Comment (vilasrk) [reply vm_battery_utility/battery_sysfsread.c]: done

**Review by balajim001 — APPROVED** [review]
LGTM

**Comment by RajaniRanjan** [comment]
LGTM

**Comment by RajaniRanjan** [comment]
LGTM

File: vm_battery_utility/battery_sysfsread.c [hunk 2]
```diff
@@ -87,10 +87,18 @@ int read_sysfs_values(char *filename, void *buf, int len, int flag)
...
-	else
-    		fscanf (fp, "%d", (int*)buf);  /* read/validate value */
+	if (flag==0) {
+	    if (fread(buf, len, 1, fp) == EOF) {
+		    fclose (fp);
+		    return -1;
+	    }
+	}
+	else {
+	    if (fscanf (fp, "%d", (int*)buf) < 0) {
```
Comment (JaikrishnaNemallapudi) [inline vm_battery_utility/battery_sysfsread.c:97]: remove space between fscanf & (

File: vm_battery_utility/battery_sysfsread.c [hunk 3]
```diff
@@ -87,10 +87,18 @@ int read_sysfs_values(char *filename, void *buf, int len, int flag)
...
-	if (flag==0)
-		fread(buf, len, 1, fp);
-	else
-    		fscanf (fp, "%d", (int*)buf);  /* read/validate value */
+	if (flag==0) {
+	    if (fread(buf, len, 1, fp) == EOF) {
+		    fclose (fp);
+		    return -1;
+	    }
+	}
```
Comment (JaikrishnaNemallapudi) [inline vm_battery_utility/battery_sysfsread.c:95]: this can be written like } else {

File: vm_battery_utility/battery_sysfsread.c [hunk 4]
```diff
@@ -87,10 +87,18 @@ int read_sysfs_values(char *filename, void *buf, int len, int flag)
 		return -ENODEV;
 	}
 
-	if (flag==0)
-		fread(buf, len, 1, fp);
-	else
-    		fscanf (fp, "%d", (int*)buf);  /* read/validate value */
+	if (flag==0) {
```
Comment (JaikrishnaNemallapudi) [inline vm_battery_utility/battery_sysfsread.c:90]: add space like if (flag == 0) {

File: vm_battery_utility/battery_sysfsread.c [hunk 5]
```diff
@@ -87,10 +87,18 @@ int read_sysfs_values(char *filename, void *buf, int len, int flag)
...
+	    if (fread(buf, len, 1, fp) == EOF) {
+		    fclose (fp);
```
Comment (JaikrishnaNemallapudi) [inline vm_battery_utility/battery_sysfsread.c:92]: maintain 4 space indentation

**Review by JaikrishnaNemallapudi — COMMENTED** [review]
comments related to indentation

**Pull request merged**